### PR TITLE
Prepare code for werror

### PR DESCRIPTION
**Pull request by evintila**
Fixed warnings for older compiler versions.

### CODE REVIEW

**Comment by evintila** [comment]
I fixed the "dereferencing type-punned pointer will break strict-aliasing rules" warnings by using memcpy.
If you guys know of a more elegant way of solving it, please share it.

**Comment by nickbroon** [comment]
Using memcpy is the correct way to copy and change the type of something accessed through a pointer.
In some cases you can use a union of the types instead, where that's meaningful and is likely to used more than once.

**Comment by evintila** [comment]
@nickbroon Thanks!

File: src/modules/ipfix/aggregator/PacketHashtable.cpp [hunk 1]
```diff
@@ -138,7 +138,9 @@ void PacketHashtable::copyDataTransportOctets(CopyFuncParameters* cfp)
 	switch (cfp->packet->ipProtocolType) {
 		case Packet::TCP:
 			ppd = reinterpret_cast<PayloadPrivateData*>(cfp->dst+cfp->efd->privDataOffset);
-			ppd->seq = ntohl(*reinterpret_cast<const uint32_t*>(p->data.netHeader+p->transportHeaderOffset+4))+plen+(p->data.netHeader[p->transportHeaderOffset+13] & 0x02 ? 1 : 0);
+			uint32_t hu32_data;
+			memcpy(&hu32_data, p->data.netHeader+p->transportHeaderOffset+4, sizeof(uint32_t));
```
Comment (ogasser) [inline src/modules/ipfix/aggregator/PacketHashtable.cpp]: should we change this to `sizeof(hu32_data)`?

File: src/modules/ipfix/aggregator/PacketHashtable.cpp [hunk 2]
```diff
@@ -200,8 +202,11 @@ void PacketHashtable::aggregateFrontPayload(IpfixRecord::Data* bucket, Hashtable
 
 	IpfixRecord::Data* dst = bucket+efd->dstIndex;
 	uint32_t seq = 0;
-	if (src->ipProtocolType==Packet::TCP)
-		seq = ntohl(*reinterpret_cast<const uint32_t*>(src->data.netHeader+src->transportHeaderOffset+4));
+	if (src->ipProtocolType==Packet::TCP) {
+		uint32_t hu32_data;
+		memcpy(&hu32_data, src->data.netHeader+src->transportHeaderOffset+4, sizeof(uint32_t));
```
Comment (ogasser) [inline src/modules/ipfix/aggregator/PacketHashtable.cpp]: should we change this to `sizeof(hu32_data)`?

File: src/modules/ipfix/aggregator/PacketHashtable.cpp [hunk 3]
```diff
@@ -1089,10 +1094,14 @@ void PacketHashtable::aggregateField(const ExpFieldData* efd, HashtableBucket* h
 			switch (p->ipProtocolType) {
 				case Packet::TCP:
 					ppd = reinterpret_cast<PayloadPrivateData*>(data+efd->privDataOffset);
-					seq = ntohl(*reinterpret_cast<const uint32_t*>(p->data.netHeader+p->transportHeaderOffset+4));
+					uint32_t hu32_data;
+					memcpy(&hu32_data, p->data.netHeader+p->transportHeaderOffset+4, sizeof(uint32_t));
```
Comment (ogasser) [inline src/modules/ipfix/aggregator/PacketHashtable.cpp]: should we change this to `sizeof(hu32_data)`?

Comment (ogasser) [reply src/modules/ipfix/aggregator/PacketHashtable.cpp]: And should we directly memcpy into `seq`?

File: src/modules/ipfix/aggregator/PacketHashtable.cpp [hunk 4]
```diff
@@ -1089,10 +1094,14 @@ void PacketHashtable::aggregateField(const ExpFieldData* efd, HashtableBucket* h
...
+					seq = hu32_data;
 
 					if (!ppd->initialized) {
-						ppd->seq = ntohl(*reinterpret_cast<const uint32_t*>(p->data.netHeader+p->transportHeaderOffset+4))+plen+(p->data.netHeader[p->transportHeaderOffset+13] & 0x02 ? 1 : 0);
+						uint32_t hu32_data;
+						memcpy(&hu32_data, p->data.netHeader+p->transportHeaderOffset+4, sizeof(uint32_t));
```
Comment (ogasser) [inline src/modules/ipfix/aggregator/PacketHashtable.cpp]: should we change this to `sizeof(hu32_data)`?

**Pull request merged**